recipes/nmma/meta.yaml

```
@@ -0,0 +1,96 @@
+{% set name = "nmma" %}
+{% set version = "0.0.7" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/nmma-{{ version }}.tar.gz
+  sha256: 9394a212c06a60e06181b2f0b78e8015d41280aac89dbc6ab19156c6f09ab3fa
+
+build:
+  entry_points:
+    - nmma_analysis=nmma.pbilby.analysis:main
+    - nmma_generation=nmma.pbilby.generation:main
+    - parallel_em_analysis=nmma.pbilby.em_analysis:main
+    - parallel_em_generation=nmma.pbilby.em_generation:main
+    - light_curve_analysis=nmma.em.analysis:main
+    - light_curve_injection_summary=nmma.em.injection_summary:main
+    - light_curve_injection_slurm_setup=nmma.em.create_injection_slurm:main
+    - light_curve_injection_condor_setup=nmma.em.create_injection_condor:main
+    - lightcurve_marginalization=nmma.em.lightcurve_marginalization:main
+    - combine_EOS=nmma.em.combine_EOS:main
+    - create_light_curve_slurm=nmma.em.create_lightcurves_slurm:main
+    - create_light_curve_condor=nmma.em.create_lightcurves_condor:main
+    - create_svdmodel=nmma.em.create_svdmodel:main
+    - light_curve_generation=nmma.em.create_lightcurves:main
+    - light_curve_detection=nmma.em.detect_lightcurves:main
+    - nmma_create_injection=nmma.eos.create_injection:main
+    - gwem_resampling=nmma.em.gwem_resampling:main
+    - gwem_resampling_condor=nmma.em.gwem_resampling_condor:main
+    - gwem_Hubble_estimate=nmma.em.gwem_Hubble_estimate:main
+    - light_curve_analysis_condor=nmma.em.analysis_condor:main
+
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  noarch: python
+
+requirements:
+  host:
+    - pip
+    - python >=3.7
+  run:
+    - arviz
+    - astropy >=4.3.1
+    - bilby >=1.2.0
+    - bilby_pipe ==1.0.5
+    - dust_extinction
+    - future
+    - matplotlib-base >=2.0
+    - numpy >=1.9
+    - pandas >=1.3.4
+    - pymultinest
+    - python >=3.7
+    - scikit-learn >=0.23.1
+    - scipy >=1.7.1
+    - sncosmo
+
+test:
+  imports:
+    - nmma
+  commands:
+    - pip check
+    - nmma_analysis --help
+    - nmma_generation --help
+    - parallel_em_analysis --help
+    - parallel_em_generation --help
+    - light_curve_analysis --help
+    - light_curve_injection_summary --help
+    - light_curve_injection_slurm_setup --help
+    - light_curve_injection_condor_setup --help
+    - lightcurve_marginalization --help
+    - combine_EOS --help
+    - create_light_curve_slurm --help
+    - create_light_curve_condor --help
+    - create_svdmodel --help
+    - light_curve_generation --help
+    - light_curve_detection --help
+    - nmma_create_injection --help
+    - gwem_resampling --help
+    - gwem_resampling_condor --help
+    - gwem_Hubble_estimate --help
+    - light_curve_analysis_condor --help
+  requires:
+    - pip
+    - mpi4py
```

Another optional dependency. It would be nice if you could check upstream for all of them instead of us playing whac-a-mole there.
```suggestion
    - mpi4py
    - nestcheck
```

---

@ocefpaf Thank you for your help. I have a bit of a hard time understanding how to replicate the exact environment one gets locally. Is the premise to try something like:

conda install -c conda-forge arviz astropy bilby bilby_pipe dust_extinction future matplotlib-base numpy pandas pymultinest scikit-learn scipy sncosmo

and then run setup.py without any requirements to be sure nothing else gets installed? 

Anyways, I am happy to sit and use GA to work through and get the right list.

---

That is OK but you don't need that to create a recipe. You can [start with `grayskull`](https://conda-forge.org/blog/posts/2020-03-05-grayskull/), that will give you a recipe that is almost "merge ready," and add the extra dependencies from the setup.py in the tests, or remove the tests that use the extras.